python/cugraph-pyg/cugraph_pyg/examples/gcn_dist_sg.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import argparse
+import tempfile
+import os
+
+from typing import Optional
+
+import torch
+import cupy
+
+import rmm
+from rmm.allocators.cupy import rmm_cupy_allocator
+from rmm.allocators.torch import rmm_torch_allocator
+
+# Must change allocators immediately upon import
+# or else other imports will cause memory to be
+# allocated and prevent changing the allocator
+rmm.reinitialize(devices=[0], pool_allocator=True, managed_memory=True)
+cupy.cuda.set_allocator(rmm_cupy_allocator)
+torch.cuda.memory.change_current_allocator(rmm_torch_allocator)
+
+import torch.nn.functional as F  # noqa: E402
+import torch_geometric  # noqa: E402
+import cugraph_pyg  # noqa: E402
+from cugraph_pyg.loader import NeighborLoader  # noqa: E402
+
+# Enable cudf spilling to save gpu memory
+from cugraph.testing.mg_utils import enable_spilling  # noqa: E402
+
+enable_spilling()
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hidden_channels", type=int, default=256)
+parser.add_argument("--num_layers", type=int, default=2)
+parser.add_argument("--lr", type=float, default=0.001)
+parser.add_argument("--epochs", type=int, default=4)
+parser.add_argument("--batch_size", type=int, default=1024)
+parser.add_argument("--fan_out", type=int, default=30)
+parser.add_argument("--tempdir_root", type=str, default=None)
+parser.add_argument("--dataset_root", type=str, default="dataset")
+parser.add_argument("--dataset", type=str, default="ogbn-products")
+
+args = parser.parse_args()
+
+wall_clock_start = time.perf_counter()
+device = torch.device("cuda")
+
+from ogb.nodeproppred import PygNodePropPredDataset  # noqa: E402
+
+dataset = PygNodePropPredDataset(name=args.dataset, root=args.dataset_root)
+split_idx = dataset.get_idx_split()
+data = dataset[0]
+
+graph_store = cugraph_pyg.data.GraphStore()
+graph_store[
+    ("node", "rel", "node"), "coo", False, (data.num_nodes, data.num_nodes)
+] = data.edge_index
+
+feature_store = cugraph_pyg.data.TensorDictFeatureStore()
+feature_store["node", "x"] = data.x
+feature_store["node", "y"] = data.y
```

Comment:
Should have  function around this to keep it cleaner and not have lingering variables like `dataset` lying around as they consume memory.  Probably return the `num_features` instead of having `dataset` live throughput the program duration.